### PR TITLE
Fix validation error when no relationships are defined

### PR DESCRIPTION
**Pull request by mjtribble**
The validation step was failing for a model that does not have relationships defined.

### CODE REVIEW

File: src/Model.js [hunk 1]
```diff
@@ -761,7 +760,7 @@ class Model {
    * @return {Array}
    */
   get attributeNames () {
-    return Object.keys(this.attributeDefinitions)
+    return this.attributeDefinitions ? Object.keys(this.attributeDefinitions) : []
```
Comment (jordanstephens) [inline src/Model.js:763]: If we can ensure that `this.attributeDefinitions` is always an object, even if empty, we shouldn't have to do this.

File: src/Model.js [hunk 2]
```diff
@@ -436,9 +436,8 @@ class Model {
 
     const attributeNames = options.attributes || this.attributeNames
     const relationshipNames = options.relationships || this.relationshipNames
-
-    const validAttributes = validateProperties(this, attributeNames, attributeDefinitions)
-    const validRelationships = validateProperties(this, relationshipNames, relationshipDefinitions)
+    const validAttributes = attributeDefinitions ? validateProperties(this, attributeNames, attributeDefinitions) : []
+    const validRelationships = relationshipDefinitions ? validateProperties(this, relationshipNames, relationshipDefinitions) : []
```
Comment (jordanstephens) [inline src/Model.js:440]: same thing here, what if we made this change instead:

```diff
  get attributeDefinitions () {
    const { type } = this.constructor
-    return schema.structure[type]
+    return schema.structure[type] || {}
  }
```

File: src/Model.js (hunk 1 above)
Comment (mjtribble) [reply src/Model.js:763]: Sure, just have to adjust it in the schema

File: src/Model.js (hunk 2 above)
Comment (mjtribble) [reply src/Model.js:440]: ok

**Comment by jordanstephens** [comment]
Make sure you rebase with master, install the latest dependencies and build again before you merge. please don't ignore `verify_build` failing. if it is failing, something is wrong.

File: src/Model.js (hunk 2 above)
Comment (jordanstephens) [reply src/Model.js:440]: Or maybe we should make sure that `schema.structure[type] = {}`.

File: spec/Model.spec.js [hunk 3]
```diff
@@ -31,6 +31,12 @@ class Note extends Model {
   @relatedToOne organization
 }
 
+class Relationshipless extends Model {
```
Comment (jtorreggiani) [inline spec/Model.spec.js:34]: I like the naming :)

**Comment by jordanstephens** [comment]
we're going to open an issue to look into perhaps defining a `@model` decorator that we can use to ensure that models are in the schema at definition time. That way we wouldn't have to depend on these fallbacks at read time.

**Comment by mjtribble** [comment]
https://github.com/artemis-ag/mobx-async-store/issues/49

**Pull request merged**